### ISSUE TITLE
systemd-boot fails to load windows if it is encrypted with bitlocker 

### ISSUE THREAD

**Issue by rokups** [systemd/systemd#3948]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)
### systemd version the issue has been seen with

231
### Used distribution

Archlinux
### In case of bug report: Expected behaviour you didn't see

Selecting auto-detected windows entry in bootloader does not boot windows when bitlocker is enabled.
### In case of bug report: Unexpected behaviour you saw

After selecting windows entry in bootloader it properly prompts to enter bitlocker password. After entering correct password systemd bootloader menu reappears instead of loading windows. If i skip systemd-boot by selecting "Microsoft Windows" boot option from UEFI firmware menu i can properly boot windows after entering bitlocker password.
### In case of bug report: Steps to reproduce the problem

Setup:
1. /dev/sda1: fat32 ESP partition
2. /dev/sda2: GNU/Linux partition
3. /dev/sda3: Windows 10 partition

I installed GNU/Linux first, then proceeded to install windows in third spare partition. This is one way i am aware of for bootloader to autodetect windows installation.

Steps:
1. Boot into windows
2. Enable bitlocker drive encryption on C:\
3. Reboot into systemd bootloader
4. Select "Microsoft Windows" option

I had to disable TPM requirement in group policy editor in order to allow enabling using bitlocker with password only. How-to [here](http://security.stackexchange.com/questions/88863/bitlocker-without-tpm-with-both-a-startup-usb-and-password#answer-88872).

**Comment by medhefgo** [comment]
First you should check if you can chainload the windows bootloader from grub.

**Comment by rokups** [comment]
I made it work by accident. It started working after i deleted `Efi\Boot\Bootx64.efi` which was systemd bootloader. I added efi boot entry to load `Efi\systemd\systemd-bootx64.efi` and booting through that if i select "Microsoft Windows" entry it asks for bitlocker password and continues to properly boot after password  is entered.

EDIT:
While this was only thing i changed that could relate (or so i think) and it is working on my workstation - doing same thing on my laptop did not fix the issue. Probably deleting said files does nothing after all. I recently cloned my workstation hdd to another one and put it in the laptop so as far as software setup goes its pretty much identical. I am again clueless why this happens. Thing is it works on workstation and does not work on laptop.

**Comment by rokups** [comment]
No longer an issue.

**Comment by jwhendy** [comment]
Followup: would it ever be possible to chainload windows *without* having to enter the bitlocker password? I've got a separate linux drive on a work laptop and don't have the passphrase. If I boot to the windows physical drive via the system boot menu (F9 on my HP Zbook 15), it's fine. If I try to chainload from `rEFInd`, it asks for the passphrase.

I'm curious if there's somewhere I can point the linux bootloader to "just do what windows would do normally." Maybe it's not possible, but there's very little information I'm finding!

**Comment by jwhendy** [comment]
@medhefgo thanks for the clarification. Really hoped to avoid f9 every time I want to choose my OS.

**Comment by IPlayZed** [comment]
For more information on this see [what Microsoft writes about this](https://learn.microsoft.com/en-us/windows/security/information-protection/bitlocker/bitlocker-countermeasures).

**Comment by jwhendy** [comment]
@IPlayZed was that a general comment for reference, or is there some specific information in there relevant to why this is not possible? It's a pretty long article, so a section to look in would be helpful.

If it was linked as the explanation of why this won't work, I would counter that rEFInd has no problems chaining on my current system, so either (a) I didn't have things setup correctly in 2018, or (b) something changed ( @medhefgo 's comment above that this is impossible is not currently correct, or never was).

I forgot all about this thread and have a new computer, so unsure of which occurred. If it's helpful, I could provide my rEFInd config if that would assist in replicating chaining in systemd-boot.

**Comment by IPlayZed** [comment]
> @IPlayZed was that a general comment for reference, or is there some specific information in there relevant to why this is not possible? It's a pretty long article, so a section to look in would be helpful.
> 
> If it was linked as the explanation of why this won't work, I would counter that rEFInd has no problems chaining on my current system, so either (a) I didn't have things setup correctly in 2018, or (b) something changed ( @medhefgo 's comment above that this is impossible is not currently correct, or never was).
> 
> I forgot all about this thread and have a new computer, so unsure of which occurred. If it's helpful, I could provide my rEFInd config if that would assist in replicating chaining in systemd-boot.

Hi, this was only general information on the topic. It doesn't work for me either when chainloading via systemd-boot.

At the very least I personally would like to take a look at your rEFInd config, specifically that which EFI is chainloaded.

**Comment by jwhendy** [comment]
@IPlayZed Awesome, thanks for clarifying :) 

Sure thing. My corporation has _drastically_ reduced usability of linux due to increased Microsoft login barricades, so I almost never boot into linux anymore, but I can take a moment to do so within a few days. Happy to provide for reference.

**Comment by poettering** [comment]
Folks, please don't add discussions to 7 year old issues that were closed 5 years ago.

current sd-boot versions can chainload windows with clean TPM state, so the whole discussion is entirely pointless by now.

**Comment by jwhendy** [comment]
@IPlayZed my refind.conf is pretty simple:
```
menuentry "arch" {
  icon		/EFI/refind/icons/os_arch.png
  volume	[uuid]
  loader	/vmlinuz-linux
  initrd	/initramfs-linux.img
  options	"luks.uuid=uuid root=UUID=uuid rootflags=compress=lzo,ssd,subvol=arch ipv6.disable_ipv6=1 add_efi_memmap rw"
}

menuentry "windows" {
  icon		/EFI/refind/icons/os_win.png
  volume	uuid
  loader	\EFI\Microsoft\Boot\bootmgfw.efi
}
```

I found I had to add some device uuids to dont_scan_volumes, as something was messing with refind when automatically finding other boot loaders. To @poettering 's point, perhaps nothing new here, and many bootloaders evolved to enable this. Sorry for the delay, I just don't boot into linux much anymore.

Sorry for the noise @poettering , was just replying to the bump and didn't look at timestamps.

**Comment by IPlayZed** [comment]
> Folks, please don't add discussions to 7 year old issues that were closed 5 years ago.
> 
> current sd-boot versions can chainload windows with clean TPM state, so the whole discussion is entirely pointless by now.

In my case it can not unfortunately this is not the case. I used [sbctl](https://github.com/Foxboron/sbctl/wiki/FAQ#option-rom)'s _Enroll UEFI vendor certificates from Microsoft into the signature database_ (option `-m`) to enroll keys in my system.

I use a common EFI partition (as otherwise Windows would fail system securiy upgrades, persumably it had something to do with the bootloader image, eventough I could technically get it to work with multiple EFI partitions) and as such systemd-boot autodetects the `bootmgfw.efi`.

The interesting part is that on an other machine, where I installed Ubuntu and have used the default vendor secboot setup (so no sbctl), this problem does not happen (but it uses Grub, so I am not sure how applicable that is here).

Sorry for dragging out an old thread, but the problem persists to me, exactly as for the OP.
